boot-mongodb-elasticsearch/src/main/java/com/example/mongoes/web/controller/RestaurantController.java

```diff
@@ -2,91 +2,76 @@
 
 import com.example.mongoes.document.Restaurant;
 import com.example.mongoes.response.GenericMessage;
+import com.example.mongoes.web.api.RestaurantApi;
 import com.example.mongoes.web.model.GradesRequest;
 import com.example.mongoes.web.model.RestaurantRequest;
 import com.example.mongoes.web.service.RestaurantService;
 import io.micrometer.core.annotation.Timed;
-import jakarta.validation.Valid;
-import jakarta.validation.constraints.Max;
-import jakarta.validation.constraints.NotBlank;
-import jakarta.validation.constraints.Pattern;
-import jakarta.validation.constraints.Size;
 import java.net.URI;
 import java.net.URLEncoder;
 import java.nio.charset.StandardCharsets;
 import org.springframework.data.elasticsearch.core.SearchPage;
 import org.springframework.http.ResponseEntity;
 import org.springframework.validation.annotation.Validated;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.PutMapping;
-import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 import reactor.core.publisher.Mono;
 
 @RestController
 @Timed
 @RequestMapping("/api/restaurant")
 @Validated
-class RestaurantController {
+class RestaurantController implements RestaurantApi {
 
     private final RestaurantService restaurantService;
 
     RestaurantController(RestaurantService restaurantService) {
         this.restaurantService = restaurantService;
     }
 
-    @GetMapping
-    Mono<ResponseEntity<SearchPage<Restaurant>>> findAllRestaurants(
-            @Valid @RequestParam(defaultValue = "10") @Max(999) int limit,
-            @RequestParam(defaultValue = "0") int offset) {
+    @Override
+    public Mono<ResponseEntity<SearchPage<Restaurant>>> findAllRestaurants(int limit, int offset) {
         return restaurantService.findAllRestaurants(offset, limit).map(ResponseEntity::ok);
     }
 
-    @GetMapping("/name/{restaurantName}")
-    Mono<ResponseEntity<Restaurant>> findRestaurantByName(
-            @PathVariable @NotBlank @Size(max = 255) @Pattern(regexp = "^[a-zA-Z0-9 .-]+$")
-                    String restaurantName) {
+    @Override
+    public Mono<ResponseEntity<Restaurant>> findRestaurantByName(String restaurantName) {
         return restaurantService
                 .findByRestaurantName(restaurantName)
                 .map(ResponseEntity::ok)
                 .defaultIfEmpty(ResponseEntity.notFound().build());
```

Comment:
@coderabbitai, how to throw `RestaurantNotFoundException` if empty, Also please let me know as per coding standards throwing from controller is better or from service